File: Holiday Assignments/Java/TransactionLogApp.java
import java.util.ArrayList;
import java.util.Scanner;

public class TransactionLogApp {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        int newBalance = 0;
        ArrayList<String> transactionList = new ArrayList<>();
        boolean continueMainMenu = true;
        while (continueMainMenu) {
            System.out.print("""
                Transaction Log App

                1. Deposit
                2. Withdraw
                3. Show Transactions
                4. Exit

                Enter here:
                """);

            int choice = input.nextInt();

            switch (choice) {

                case 1 ->  {
            boolean continueDeposit = true;
		    while (continueDeposit){
                    System.out.print("Enter deposit amount: ₦");
                    int amount = input.nextInt();
		    if (amount > 0){
                        newBalance += amount;

                        String transaction = "Deposited: ₦" + amount + " | New Balance: ₦" + newBalance;

                        transactionList.add(transaction);
                        System.out.println(transaction);
                        System.out.println();
			break;
		    }
		    else{
			System.out.println("Invalid amount");
		    }
                }}

                case 2 ->  {
            boolean continueWithdraw = true;
		    while (continueWithdraw){
                    System.out.print("Enter withdrawal amount: ₦");
                    int amount = input.nextInt();
		    if (amount > 0){
                if (amount > newBalance){
                    System.out.println("Insufficient funds. Available balance: ₦" + newBalance);
                    continue;
                }
                else{
                         newBalance -= amount;

                        String transaction = "Withdrew: ₦" + amount + " | New Balance: ₦" + newBalance;

                        transactionList.add(transaction);
                        System.out.println(transaction);
                        System.out.println();
			            break;
                }
			}
		    else{
			System.out.println("Invalid amount");
		    }}
                }

                case 3 ->  {
                    if (transactionList.isEmpty()) {
                        System.out.println("No transactions yet.");
                    } else {
                        for (int i = 0; i < transactionList.size(); i++) {
                            System.out.println((i + 1) + ". " + transactionList.get(i));
                        }
                    }
                    System.out.println();
                }

                case 4 -> {
                    System.out.println("Final Balance: ₦" + newBalance);
                    input.close();
                    return;
                }

                default -> 
                    System.out.println("Invalid option, please try again.");
            }
        }
    }
}

File: Holiday Assignments/Python/transaction_log_app.py
def deposit(x, y):
    
    new_balance += amount
    return new_balance
    
def transactionappend(transaction):
    transaction_list.append(transaction)
    return transaction_list
    
    
new_balance = 0
transaction_list = []
    
def transactionLogApp():
    
    continue_main_menu = True
    while continue_main_menu:
                print(
"""
Transaction Log App

1. Deposit
2. Withdraw
3. Show Transactions
4. Exit

Enter here: """
                        )

                main_menu_input = int(input())
                match main_menu_input:
                    
                    case 1 :
                        continue_deposit = True
                        while continue_deposit:
                            amount = float(input("Enter deposit amount: "))
                            if amount > 0:
                                deposit(new_balance, amount)
                                transaction = ("Deposited: ₦", amount, "| New Balace: ₦", new_balance)
                                print("Transaction", len(transaction_list))
                                print(f"Deposited: ₦{amount} | New Balace: ₦{new_balance}")
                            else:
                                 print("Invalid amount")

                    case 2 :
                        continue_withdraw = True
                        while continue_withdraw:
                            amount = int(input("Enter withdrawal amount: "))
                            if amount > 0:
                                if amount > new_balance:
                                    print("Insufficient funds. Available balance: ₦", new_balance, sep = "")
                                    
                                else:
                                    new_balance -= amount
                                    transaction_list.append(("Withdrew: ₦", amount, "| New Balace: ₦", new_balance))
                                    print("Transaction", len(transaction_list))
                                    print(f"Withdrew: ₦{amount} | New Balace: ₦{new_balance}")
                                    break
                            else:
                                 print("Invalid amount")

                    case 3 :
                        if transaction_list == []:
                            print("No transactions yet.")
                        else:
                            for index, (action, amount, text, balance) in enumerate(transaction_list):
                                print(f"{index + 1}. {action}{amount} {text}{balance}")

                            
                    case 4 :
                        print("Final Balance: ₦", new_balance, sep = "")
                        break 
                    
                    case _ :
                        print("Invalid, Please try again.")
transactionLogApp()




















Bool (True or False)














